server/src/main/kotlin/suwayomi/tachidesk/manga/impl/update/Updater.kt

```diff
@@ -123,54 +124,85 @@ class Updater : IUpdater {
      */
     private fun updateStatus(
         jobs: List<UpdateJob>,
-        running: Boolean,
+        running: Boolean? = null,
         categories: Map<CategoryUpdateStatus, List<CategoryDataClass>>? = null,
         skippedMangas: List<MangaDataClass>? = null,
     ) {
+        val isRunning =
+            running
+                ?: jobs.any { job ->
+                    job.status == JobStatus.PENDING || job.status == JobStatus.RUNNING
+                }
         val updateStatusCategories = categories ?: _status.value.categoryStatusMap
         val tmpSkippedMangas = skippedMangas ?: _status.value.mangaStatusMap[JobStatus.SKIPPED] ?: emptyList()
-        _status.update { UpdateStatus(updateStatusCategories, jobs, tmpSkippedMangas, running) }
+        _status.update { UpdateStatus(updateStatusCategories, jobs, tmpSkippedMangas, isRunning) }
     }
 
     private fun getOrCreateUpdateChannelFor(source: String): Channel<UpdateJob> {
         return updateChannels.getOrPut(source) {
             logger.debug { "getOrCreateUpdateChannelFor: created channel for $source - channels: ${updateChannels.size + 1}" }
-            createUpdateChannel()
+            createUpdateChannel(source)
         }
     }
 
-    private fun createUpdateChannel(): Channel<UpdateJob> {
+    private fun createUpdateChannel(source: String): Channel<UpdateJob> {
         val channel = Channel<UpdateJob>(Channel.UNLIMITED)
         channel.consumeAsFlow()
             .onEach { job ->
                 semaphore.withPermit {
-                    updateStatus(
-                        process(job),
-                        tracker.any { (_, job) ->
-                            job.status == JobStatus.PENDING || job.status == JobStatus.RUNNING
-                        },
-                    )
+                    process(job)
                 }
             }
-            .catch { logger.error(it) { "Error during updates" } }
+            .catch {
+                logger.error(it) { "Error during updates (source: $source)" }
+                handleChannelUpdateFailure(source)
+            }
+            .onCompletion { updateChannels.remove(source) }
             .launchIn(scope)
         return channel
     }
 
-    private suspend fun process(job: UpdateJob): List<UpdateJob> {
+    private fun handleChannelUpdateFailure(source: String) {
+        val isFailedSourceUpdate = { job: UpdateJob ->
+            val isForSource = job.manga.sourceId == source
+            val hasFailed = job.status == JobStatus.FAILED
+
+            isForSource && hasFailed
+        }
+        val failAllUpdatesForSource = {
```

Comment:
I don't think this Lamba is needed, can you remove it and just add a comment?

Reply:
removed